visualizer/data-tree.js

```diff
@@ -95,28 +96,65 @@ class DataTree {
     }
   }
 
-  getFilteredStackSorter () {
-    const exclude = this.exclude
-
-    function getValue (node) {
-      if (exclude.has(node.type)) {
-        // Value of hidden frames is the sum of their visible children
-        return node.children ? node.children.reduce((acc, child) => {
-          return acc + getValue(child)
-        }, 0) : 0
-      }
-
-      // d3-fg sets `value` to 0 to hide off-screen nodes.
-      // there's no other property to indicate this but the original value is stored on `.original`.
-      if (node.value === 0 && typeof node.original === 'number') {
-        return node.original
-      }
-      return node.value
+  computeGroupedSortValues () {
+    this.groupedSortValues = new Map()
+
+    function getTypeKey (node) {
+      return `${node.category}:${node.typeTEMP !== undefined ? node.typeTEMP : node.type}`
+    }
+
+    const walk = (node) => {
+      if (!node.children) return
+      const group = Object.create(null)
+      node.children.forEach((child) => {
+        const type = getTypeKey(child)
+        const value = this.getSortValue(child)
```

Comment:
We're getting some type key / sort value mismatches here, because getSortValue is recursing up the tree ignoring excluded frames, but getTypeKey gets the type of the immediate children, regardless of exclusion. 

So, if a child is excluded, and the grandchild isn't, we get the value of the visible grandchild against the key of the excluded child.

So we should probably return key/value pairs in getSortValue which would then get merged by the existing `if (type in group) {  group[type] += value...` in computeGroupedSortValues walk()?

Reply:
Maaaaaaybe. If we merge sort values from grandchildren, those grandchildren will not necessarily end up next to other children of the same type.

assuming an unfiltered tree like:
```
          | c | d | d | c | d |
| c | c | hidden              |
```

when `hidden` is filtered away, would become:
```
| c | c | d | d | d | c | c |
```

but maybe that's still better/at least more predictable than the current situation where a hot hidden frame can move some very small stacks to the left of larger visible stacks. I'll play with it a bit…

(it would be kinda nice if we could totally ignore hidden frames but then we may need to actually remove hidden nodes from the trees, which sounds expensive…)